pipeline/src/pipeline.rs

```diff
@@ -724,15 +724,16 @@ impl<T: FieldElement> Pipeline<T> {
     }
 
     fn maybe_write_proof(&self, proof_result: &ProofResult<T>) -> Result<(), Vec<String>> {
-        if let Some(constraints_serialization) = &proof_result.constraints_serialization {
+        // TODO: this must go inside the backend...
```

Comment:
I disagree with this. IMO the user needs to access the artifact and do whatever it needs with it